byok documentation (#6634)
camunda/camunda-docs: docs/components/saas/byok/cost-and-troubleshooting.md

Author: @psetzer-camunda Do you still have additional revisions incoming?

Reviewer: @psetzer-camunda is on parental leave until the 22th of October. So we won't have update by him. I think it's okay for now.

---
id: cost-and-troubleshooting

> title: "Cost implications and troubleshooting"

Reviewer 2: Hard to get context.
External Encryption Keys cost implications and troubelshooting

description: "Learn more about cost implications and troubleshooting."
---

> Using BYOK with AWS KMS incurs costs directly in your AWS account. Camunda 8.8 does not charge for the BYOK feature itself, but you are responsible for all AWS KMS usage.

Reviewer: ```suggestion
Using external encryption keys with AWS KMS incurs costs directly in your AWS account. Camunda does not charge for the feature itself, but you are responsible for all AWS KMS usage.
```

> Using external encryption keys with **Amazon KMS** incurs costs directly in your Amazon account. Camunda does not charge for the feature itself, but you are responsible for all Amazon KMS usage.

Reviewer 3: ```suggestion
Using external encryption keys with **AWS KMS** incurs costs directly in your AWS account. Camunda does not charge for the feature itself, but you are responsible for all AWS KMS usage.
```

> ## Cost implications

Reviewer 2: I suggest moving this section under the main BYOK page, as it is an important part of the configuration.

Reviewer 2: Cost and troubleshooting do not seem to be related to each other, ie my main point. I think moving cost to main article would be more fitting for this info.

| Cost type       | Description                                                              | Notes                                                      |
| --------------- | ------------------------------------------------------------------------ | ---------------------------------------------------------- |

> | KMS key storage | Monthly charge for each KMS key                                          | Depends on Amazon region and key type                      |

Reviewer 3: ```suggestion
| KMS key storage | Monthly charge for each KMS key                                          | Depends on AWS region and key type                         |
```

| API requests    | Charges for KMS API calls (Encrypt, Decrypt, GenerateDataKey, ReEncrypt) | Costs increase with frequent operations                    |
| CloudTrail logs | Charges for storing and accessing CloudTrail events                      | Includes encryption/decryption activity by Camunda cluster |

:::warning Cost responsibility

> You are responsible for monitoring AWS KMS usage and associated costs. Camunda does not bill for KMS or CloudTrail usage.

Reviewer: ```suggestion
You are responsible for monitoring AWS KMS usage and associated costs.
```

:::

### Cost optimization tips

- Use separate keys only when necessary to avoid extra storage fees.
- Aggregate audit logging to reduce frequent API calls.
- Review CloudTrail retention settings to balance compliance and storage cost.

## Troubleshooting

The following table summarizes common issues customers may encounter when using external encryption keys and recommended actions:

| Issue                             | Possible cause                                                   | Resolution                                                                                                                                                                                                                         |
| --------------------------------- | ---------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Cluster cannot access KMS key     | Key policy does not grant the Camunda Cluster Amazon Role access | Update KMS key policy to include the correct Amazon Role ARN from the Camunda Console                                                                                                                                              |

Reviewer 2: Cluster cannot access KMS key - is there an error message that customer can see? What would be the issue that customers face?

| Encryption/decryption errors      | Key is disabled, deleted, or in the wrong region                 | Re-enable or restore key, or create a new key in the same region as the Camunda cluster                                                                                                                                            |

> | CloudTrail does not show activity | CloudTrail not enabled or log retention insufficient             | Enable CloudTrail in the Amazon region where the cluster resides and persist logs beyond the default 90 days. See [View CloudTrail events](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/view-cloudtrail-events.html) |

Reviewer 2: Enable - enable (nit)

| Key rotation issues               | Cluster encryption update not supported                          | Create a new key and associate it with a new cluster if rotation is required. Verify encryption settings before using the new key.                                                                                                 |

:::note Support

> If issues persist after checking key policies, region, and key status, contact Amazon support for KMS-related troubleshooting. For Camunda-specific issues with cluster provisioning, contact Camunda support.

Reviewer 2: Can we add support link https://camunda.com/services/support-guide/
